### PR TITLE
Fix returned fields in Get-AzureRmSubscription docs

### PR DESCRIPTION
**Pull request by SteGriff**
## Description

Fixed the examples of returned fields in Get-AzureRmSubscription. Added an extra piped statement to Example 4 because it would not naturally work with the newly renamed fields.

Fixes issue #4935

---

This checklist is used to make sure that common guidelines for a pull request are followed. You can find a more complete discussion of PowerShell cmdlet best practices [here](https://msdn.microsoft.com/en-us/library/dd878270(v=vs.85).aspx).

- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md).**
- [x] **If changes were made to any cmdlet, the XML help was regenerated using the [platyPSHelp module](https://github.com/Azure/azure-powershell/blob/preview/documentation/help-generation.md).**
- [x] **If any large changes are made to a service, they are reflected in the respective [change log](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#updating-the-change-log).**

### [General Guidelines](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/preview/documentation/cleaning-up-commits.md).
- [x] The pull request does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/preview/documentation/breaking-changes/breaking-changes-definition.md) (unless a major version change occurs in the assembly and module).

### [Testing Guidelines](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#testing-guidelines)
- [x] Pull request includes test coverage for the included changes.
- [x] PowerShell scripts used in tests should do any necessary setup as part of the test or suite setup, and should not use hard-coded values for locations or existing resources.

### [Cmdlet Signature Guidelines](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#cmdlet-signature-guidelines)
- [x] New cmdlets that make changes or have side effects should implement `ShouldProcess` and have `SupportShouldProcess=true` specified in the cmdlet attribute. You can find more information on `ShouldProcess` [here](https://github.com/Azure/azure-powershell/wiki/PowerShell-Cmdlet-Design-Guidelines#supportsshouldprocess).
- [x] Cmdlet specifies `OutputType` attribute if any output is produced - if the cmdlet produces no output, it should implement a `PassThru` parameter.

### [Cmdlet Parameter Guidelines](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#cmdlet-parameter-guidelines)
- [x] Parameter types should not expose types from the management library - complex parameter types should be defined in the module.
- [x] Complex parameter types are discouraged - a parameter type should be simple types as often as possible. If complex types are used, they should be shallow and easily creatable from a constructor or another cmdlet.
- [x] Cmdlet parameter sets should be mutually exclusive - each parameter set must have at least one mandatory parameter not in other parameter sets.

### CODE REVIEW

**Comment by SteGriff** [comment]
Hi @cormacpayne, how does this look?

File: src/ResourceManager/Profile/Commands.Profile/help/Get-AzureRmSubscription.md [hunk 1]
```diff
@@ -63,25 +66,30 @@ current account.
...
+State    : Enabled
 ```
 
 This command gets all subscriptions in the current tenant that are
 authorized for the current user.
 
 ### Example 4: Change the current context to use a specific subscription
 ```
-PS C:\>Get-AzureRmSubscription -SubscriptionId "xxxx-xxxx-xxxx-xxxx" -TenantId "yyyy-yyyy-yyyy-yyyy" | Set-AzureRmContext
+PS C:\>Get-AzureRmSubscription -SubscriptionId "xxxx-xxxx-xxxx-xxxx" -TenantId "yyyy-yyyy-yyyy-yyyy" | Select @{n='SubscriptionId';e={$_.Id}} | Set-AzureRmContext
```
Comment (cormacpayne) [inline src/ResourceManager/Profile/Commands.Profile/help/Get-AzureRmSubscription.md]: @SteGriff the `Set-AzureRmContext` cmdlet accepts a `PSAzureSubscription` object from the pipeline for the `SubscriptionObject` parameter. Any reason you want to change it to pass the `SubscriptionId` to `Set-AzureRmContext` instead? (this is just out of curiosity; I actually didn't know about the `Select @{n=<name>;e=<expression>}` trick ☺️ )

**Comment by cormacpayne** [comment]
@azuresdkci add to whitelist

File: src/ResourceManager/Profile/Commands.Profile/help/Get-AzureRmSubscription.md (hunk 1 above)
Comment (SteGriff) [reply src/ResourceManager/Profile/Commands.Profile/help/Get-AzureRmSubscription.md]: My experience of the command as written is that it doesn't work. Here's what happens for me, maybe you can verify?
~~~~
# Outputs my context unchanged. The subscription is not switched, no error:
Get-AzureRmSubscription -SubscriptionId xxxx -TenantId yyyy | Set-AzureRmContext

# Capturing the object specifically and passing it as context throws error:
$ctx = Get-AzureRmSubscription -SubscriptionId xxxx -TenantId yyyy
Set-AzureRmContext -Context $ctx

Set-AzureRmContext : Cannot bind parameter 'Context'. Cannot convert the "8d6daea2-6038-4360-89a9-9944afa3ff6d" 
value of type "Microsoft.Azure.Commands.Profile.Models.PSAzureSubscription" to type 
"Microsoft.Azure.Commands.Profile.Models.PSAzureContext".
At line:1 char:29
+ Set-AzureRmContext -Context $ctx
+                             ~~~~
    + CategoryInfo          : InvalidArgument: (:) [Set-AzureRmContext], ParameterBindingException
    + FullyQualifiedErrorId : CannotConvertArgumentNoMessage,Microsoft.Azure.Commands.Profile.SetAzureRMContextCo 
   mmand
~~~~

Maybe I'm being stupid but this pipe has never worked for me. I have also tried `Select-AzureRmSubscription` at the end and it behaves the same. That's why I've gotten the habit of extracting the `SubscriptionId` specifically.

Perhaps we're straying into the territory of another issue and I should just put the sample the way you have said?

Comment (cormacpayne) [reply src/ResourceManager/Profile/Commands.Profile/help/Get-AzureRmSubscription.md]: Ah, so `Get-AzureRmSubscription` doesn't return a `PSAzureContext`, it returns a `PSAzureSubscription`, so you'll need to run the following:

Comment (markcowl) [reply src/ResourceManager/Profile/Commands.Profile/help/Get-AzureRmSubscription.md]: Also, this works:

```
PS C:\Users\markcowl> Get-AzureRmSubscription | Set-AzureRmContext


Name             : [markcowl@microsoft.com, xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxx]
Account          : markcowl@microsoft.com
SubscriptionName : node
TenantId         : xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxx
Environment      : AzureCloud

Name             : [markcowl@microsoft.com, xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxx]
Account          : markcowl@microsoft.com
SubscriptionName : Azure SDK CI
TenantId         : xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxx
Environment      : AzureCloud

Name             : [markcowl@microsoft.com, xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxx]
Account          : markcowl@microsoft.com
SubscriptionName : Node CLI Test
TenantId         : xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxx
Environment      : AzureCloud
```

Comment (markcowl) [reply src/ResourceManager/Profile/Commands.Profile/help/Get-AzureRmSubscription.md]: PowerShell is smart enough to accept pipeline input for a subscription or a context for this cmdlet

**Comment by SteGriff** [comment]
@cormacpayne Thanks for the feedback! I think I have amended the PR correctly (my first time amending a PR on here) 😓

**Comment by cormacpayne** [comment]
@SteGriff no worries at all! We appreciate the fix ☺️ The changes in this PR look good to me, so I'll go ahead and merge it later today or tomorrow when we are finished with a release we're doing (need to make sure the release branch gets merged back into `preview` before merging anything into `preview`).

**Pull request merged**